src/transaction.ts

```diff
@@ -1326,12 +1324,30 @@ export class Snapshot extends EventEmitter {
       this._inlineBeginStarted = true;
       return makeRequest;
     }
-    return (resumeToken?: ResumeToken): Readable =>
-      this._idWaiter.once('notify', () =>
+
+    // Queue subsequent requests.
+    return (resumeToken?: ResumeToken): Readable => {
+      const streamProxy = new Readable({
+          read() {
+          }
+      });
+
+      this._waitingRequests.push(() => {
         makeRequest(resumeToken)
-          .on('data', chunk => this._idWaiter.emit('data', chunk))
-          .once('end', () => this._idWaiter.emit('end'))
-      );
+        .on('data', (chunk) => streamProxy.emit('data', chunk))
+        .on('end', () => streamProxy.emit('end'));
+      });
+
+      return streamProxy;
+    };
+  }
+
+  _releaseWaitingRequests() {
+    this._inlineBeginStarted = false;
```

Comment:
I don't think we should be resetting this here. It seems that it goes well even if we do reset it here, but it seems a bit strange, as `inlineBeginStarted` is still true for this transaction.

Reply:
This was an accidental push. I was trying something. It should not be here.